src/App.tsx

```diff
@@ -15,48 +27,100 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: Reordering,
+) {
+  const visibleGoods = [...goods];
+
+  switch (sortType) {
+    case SortType.Alphabet:
+      visibleGoods.sort((goodA, goodB) => goodA.localeCompare(goodB));
+      break;
+
+    case SortType.Length:
+      visibleGoods.sort((goodA, goodB) => goodA.length - goodB.length);
+      break;
+
+    default:
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortBy, setSortBy] = useState(SortType.None);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const reverseToggle = () => setIsReversed(prev => (!prev));
```

Comment:
Well, this one must look like I'm a grammar nazi, but actually what this method does is toggling the reverse, not reversing the toggle 😄
The name of the function normally starts with a verb
```suggestion
  const toggleReverse = () => setIsReversed(prev => (!prev));
```